src/openrct2/scenario/scenario.c

```diff
@@ -967,12 +966,11 @@ static void scenario_objective_check_replay_loan_and_park_value()
 
 static void scenario_objective_check_monthly_food_income()
 {
-    money32 *expenditureLastMonth = &gExpenditureTable[1 * RCT_EXPENDITURE_TYPE_COUNT];
     sint32 lastMonthProfit =
-        expenditureLastMonth[RCT_EXPENDITURE_TYPE_SHOP_SHOP_SALES] +
-        expenditureLastMonth[RCT_EXPENDITURE_TYPE_SHOP_STOCK] +
-        expenditureLastMonth[RCT_EXPENDITURE_TYPE_FOODDRINK_SALES] +
-        expenditureLastMonth[RCT_EXPENDITURE_TYPE_FOODDRINK_STOCK];
+        gExpenditureTable[1][RCT_EXPENDITURE_TYPE_SHOP_SHOP_SALES] +
+        gExpenditureTable[1][RCT_EXPENDITURE_TYPE_SHOP_STOCK] +
+        gExpenditureTable[1][RCT_EXPENDITURE_TYPE_FOODDRINK_SALES] +
+        gExpenditureTable[1][RCT_EXPENDITURE_TYPE_FOODDRINK_STOCK];
```

Comment:
What was wrong with the previous way? I mean if you want to remove the multiply then it would be good if it returned a struct or array for the month you want.

E.g.
```
money32 *expenditureLastMonth = GetExpenditures(1);
```

Reply:
> What was wrong with the previous way?

Oh, come on, you can't be serious. The old way was using a 224-element array for something that was an array of 16 months of 14 types of expenditure.

Reply:
I am not taking about the way it is stored just that you are repeating the first array access multiple times.

Reply:
Why is that a bad thing? (This is not meant rhetorical.)
I changed it because Xcode wouldn't compile if I didn't make changes to how it worked, and to me this makes more sense anyway.

Reply:
You're basically looking up the relevant array (pointer) multiple times in a row. The compiler may be smart enough to work its way around it with `-O2`, but I agree it's probably better to save the pointer yourself. Both because it makes the flow more explicit, and you don't have to rely on optimisation behaviour.